### PR TITLE
WG opening listing sorted in chronological order (#2513)

### LINKED ISSUE #2513: [USERSNAP] Working Groups: Openings not listed chronologically
Add sorting by date for openings:

Default = Newest
Other option: Oldest 

_____
https://github.com/Joystream/pioneer/issues/2294

**Taken at**: https://pioneer-2.vercel.app/#/working-groups/openings
**Fields**:
- Houston, we have a problem.: Working Groups: Openings not listed chronologically
- Give us some details: SSIA
- SSIA
- Discrod Handle: tomato#9464

SCREENSHOT
*[Download original image](https://resources.usersnap.com/company/f9f719df-a929-4461-bb6e-46dbfbceffb7/datapoint_screenshot/4a656bf4-1dfa-4d36-83b5-6db103b1d9f8-annotated_9ac25d72-3f0a-4ac2-910e-169166ab6799.png?etag=0f02df417c0bda05a570b033ccea6063)*

**Browser**: Chrome 99 (Windows 10)
**Screen size**: 3440x1440
**Browser size**: 1850x1176
**Labels**: bug-non-critical
**[Open #108 in Usersnap](https://app.usersnap.com/l/feedback/bdaf42e7-44c8-4dc0-b191-918799a7b7c5?utm_medium=integration&utm_campaign=integration&utm_content=open)**

Powered by **[Usersnap](https://usersnap.com/?utm_source=product&amp;utm_medium=poweredbylink&amp;utm_campaign=github_entry)**.

### PR DESCRIPTION
**Pull request by chrlschwb**
fixes #2513
I have closed the old PR, https://github.com/Joystream/pioneer/pull/3862.
This is the new PR based on the recent dev branch.

### CODE REVIEW

**Review by traumschule — APPROVED** [review]

File: packages/ui/src/working-groups/queries/__generated__/workingGroups.generated.tsx [hunk 1]
```diff
@@ -722,7 +722,7 @@ export type CountWorkingGroupWorkersQuery = {
 
 export type GetWorkingGroupOpeningsQueryVariables = Types.Exact<{
   where?: Types.InputMaybe<Types.WorkingGroupOpeningWhereInput>
-  order?: Types.InputMaybe<Array<Types.WorkingGroupOpeningOrderByInput> | Types.WorkingGroupOpeningOrderByInput>
+  order?: Types.InputMaybe<Types.WorkingGroupOpeningOrderByInput>
```
Comment (thesan) [inline packages/ui/src/working-groups/queries/__generated__/workingGroups.generated.tsx:725]: This file is generated, it shouldn't be edited manually, or else the changes will be overridden every time someone runs `yarn workspace @joystream/pioneer queries:generate`

File: packages/ui/src/working-groups/hooks/useOpeningsPagination.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ export const useOpeningsPagination = ({ groupId, type, page = 1 }: UseOpeningsPa
     limit: OPENINGS_PER_PAGE,
     offset: (page - 1) * OPENINGS_PER_PAGE,
     where,
-    order: [WorkingGroupOpeningOrderByInput.CreatedAtDesc, WorkingGroupOpeningOrderByInput.RuntimeIdDesc],
+    order: WorkingGroupOpeningOrderByInput.CreatedAtDesc,
```
Comment (thesan) [inline packages/ui/src/working-groups/hooks/useOpeningsPagination.ts:30]: So this change should be:
```suggestion
    order: [WorkingGroupOpeningOrderByInput.CreatedAtDesc],
```
But I don't understand why does this solve the issue. Given that there is an issue to begin with: see my comment on [#2513](https://github.com/Joystream/pioneer/issues/2513#issuecomment-1437373412)

**Review by thesan — CHANGES_REQUESTED** [review]

File: packages/ui/src/working-groups/hooks/useOpeningsPagination.ts (hunk 2 above)
Comment (chrlschwb) [reply packages/ui/src/working-groups/hooks/useOpeningsPagination.ts:30]: The goal of issue https://github.com/Joystream/pioneer/issues/2513 is to sort openings according to "created date", not by "end date".
It's true that created date is displayed nowhere in the UI, but that's irrelevant to the issue. Maybe we can add "created date" of openings as a new column.

**Comment by thesan** [comment]
@chrlschwb 

> The goal of issue #2513 is to sort openings according to "created date", not by "end date". It's true that created date is displayed nowhere in the UI, but that's irrelevant to the issue. Maybe we can add "created date" of openings as a new column.

I understand that but in the comment I was questioning weither there was an issue in the first place. Could you link to an example please ?

Also to fix the CI please reverse the change on `packages/ui/src/working-groups/queries/__generated__/workingGroups.generated.tsx`

File: packages/ui/src/working-groups/queries/__generated__/workingGroups.generated.tsx (hunk 1 above)
Comment (traumschule) [inline packages/ui/src/working-groups/queries/__generated__/workingGroups.generated.tsx:725]:
```suggestion
  order?: Types.InputMaybe<Array<Types.WorkingGroupOpeningOrderByInput> | Types.WorkingGroupOpeningOrderByInput>
```

**Review by traumschule — CHANGES_REQUESTED** [review]
```
| ERROR in src/working-groups/hooks/useOpeningsPagination.ts:30:5
--
13:16:25.045 | \| TS2322: Type 'WorkingGroupOpeningOrderByInput.CreatedAtDesc[]' is not assignable to type 'InputMaybe<WorkingGroupOpeningOrderByInput> \| undefined'.
13:16:25.045 | \|   Type 'WorkingGroupOpeningOrderByInput.CreatedAtDesc[]' is not assignable to type 'WorkingGroupOpeningOrderByInput.UpdatedAtDesc'.
13:16:25.045 | \|     28 \|     offset: (page - 1) * OPENINGS_PER_PAGE,
13:16:25.045 | \|     29 \|     where,
13:16:25.045 | \|   > 30 \|     order: [WorkingGroupOpeningOrderByInput.CreatedAtDesc],
13:16:25.045 | \|        \|     ^^^^^
13:16:25.045 | \|     31 \|   }
13:16:25.045 | \|     32 \|
13:16:25.045 | \|     33 \|   const { data, loading, error } = useGetWorkingGroupOpeningsQuery({ variables })
13:16:25.045 | \|
13:16:25.045 | \| webpack 5.64.3 compiled with 1 error in 106165 ms
```

File: packages/ui/src/working-groups/hooks/useOpeningsPagination.ts [hunk 3]
```diff
@@ -27,7 +27,7 @@ export const useOpeningsPagination = ({ groupId, type, page = 1 }: UseOpeningsPa
     limit: OPENINGS_PER_PAGE,
     offset: (page - 1) * OPENINGS_PER_PAGE,
     where,
-    order: [WorkingGroupOpeningOrderByInput.CreatedAtDesc, WorkingGroupOpeningOrderByInput.RuntimeIdDesc],
+    order: [WorkingGroupOpeningOrderByInput.CreatedAtDesc],
```
Comment (thesan) [inline packages/ui/src/working-groups/hooks/useOpeningsPagination.ts:30]: If I understand correctly this https://github.com/Joystream/pioneer/issues/2513#issuecomment-1575299539, the issue is that the openings are ordered by ascending creation date. So this line (`order: [WorkingGroupOpeningOrderByInput.CreatedAtDesc, WorkingGroupOpeningOrderByInput.RuntimeIdDesc]` looks fine to me, but the problem is still not solved on [the preview branch `/working-groups/openings`](https://dao-git-fork-chrlschwb-fix-2513-sortinglist-joystream.vercel.app/#/working-groups/openings):
image

So I'm guessing that maybe this `useOpeningsPagination` is used to query the openings on a different page. E.g on the WG past openings page the order is correct even [in production](https://pioneerapp.xyz/#/working-groups/builders).

In conclusion I think this change should be reversed and either the hook used to query `/working-groups/openings` should be fixed or an even better solution would be to reuse the same hook to query openings everywhere.

Also please get reviews from BWG workers before setting `jsg-code-review`.

**Review by thesan — CHANGES_REQUESTED** [review]

**Comment by traumschule** [comment]
Merged in 3.9.0
